lib/tasks/data_hygiene/publishing_delay_reporter.rb

```diff
@@ -1,13 +1,39 @@
 class Tasks::DataHygiene::PublishingDelayReporter
+  STATISTICS_TYPES = %w(
+    national_statistics
+    official_statistics
+    national_statistics_announcement
+    official_statistics_announcement
+  ).freeze
+
   def report
     now = Time.zone.now
 
     log_entries = ScheduledPublishingLogEntry.where(created_at: (now - 1.day)..now)
-    delays = log_entries.map(&:delay_in_milliseconds)
+    stats_reporter = StatsReporter.new(log_entries)
+
+    STATISTICS_TYPES.each do |stats_type|
+      stats_reporter.report(stats_type) { |doc_type| doc_type == stats_type }
+    end
+
+    stats_reporter.report("other_document_types") { |doc_type| STATISTICS_TYPES.exclude?(doc_type) }
+
+    stats_reporter.report("all_document_types") { true }
+  end
+
+  class StatsReporter
+    attr_reader :log_entries
+
+    def initialize(log_entries)
+      @log_entries = log_entries
+    end
+
+    def report(namespace)
```

Comment:
This is extremely minor, but could we avoid the slightly awkward `{ true }` argument when we want to report all the docs?  Something like:

```ruby
def report(namespace, &filter)
  delays = log_entries.select(&filter).map(&:delay_in_milliseconds)
  # rest of method...
end
```

Seems that if you pass a nil block to select it becomes a no-op.  I found this out in irb as follows:

```ruby
def foo(&block)
  [1,2,3].select(&block).map { |x| puts x }
end

irb(main):005:0> foo { true }
1
2
3
=> [nil, nil, nil]
irb(main):006:0> foo
1
2
3
=> [nil, nil, nil]
```

I thought we'd need to do a conditional, but it seems `select` is happy with a nil block and acts like `each` because every element is selected.  I'm like 50:50 on this being clear though, as I found it a little surprising, so entirely up to you.

Reply:
I think that is neater without the `{ true }`. 👍 

Hopefully the purpose of `stats_reporter.report("all_document_types")` is clear enough, and it is unit tested.